src/main/java/org/mastodon/mamut/clustering/ClusterRootNodesController.java

```diff
@@ -245,10 +266,12 @@ private List< BranchSpotTree > getRoots()
 			try
 			{
 				BranchSpotTree branchSpotTree = new BranchSpotTree( rootBranchSpot, cropEndTime );
+				String tagValue = ClusterUtils.getTagValue( model, rootBranchSpot, tagSet );
+				BranchSpotTreeWithExtraLabel tree = new BranchSpotTreeWithExtraLabel( branchSpotTree, tagValue );
```

Comment:
I would suggest: `BranchSpotTree tree = tagValue == null ? branchSpotTree : new BranchSpotTreeWithExtraLabel( branchSpotTree, tagValue );`

This would allow to simplify the class BranchSpotTreeWithExtraLabel. Because inside the class tagValue would allways be non null.

Reply:
Good idea. Changed it as suggested.